android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -381,57 +479,7 @@ protected RNCWebView createRNCWebViewInstance(ThemedReactContext reactContext) {
   @TargetApi(Build.VERSION_CODES.LOLLIPOP)
   protected WebView createViewInstance(ThemedReactContext reactContext) {
     RNCWebView webView = createRNCWebViewInstance(reactContext);
-    webView.setWebChromeClient(new WebChromeClient() {
-      @Override
-      public boolean onConsoleMessage(ConsoleMessage message) {
-        if (ReactBuildConfig.DEBUG) {
-          return super.onConsoleMessage(message);
-        }
-        // Ignore console logs in non debug builds.
-        return true;
-      }
-
-
-    @Override
-    public void onProgressChanged(WebView webView, int newProgress) {
-        super.onProgressChanged(webView, newProgress);
-        WritableMap event = Arguments.createMap();
-        event.putDouble("target", webView.getId());
-        event.putString("title", webView.getTitle());
-        event.putBoolean("canGoBack", webView.canGoBack());
-        event.putBoolean("canGoForward", webView.canGoForward());
-        event.putDouble("progress", (float)newProgress/100);
-        dispatchEvent(
-                  webView,
-                  new TopLoadingProgressEvent(
-                      webView.getId(),
-                      event));
-    }
-
-      @Override
-      public void onGeolocationPermissionsShowPrompt(String origin, GeolocationPermissions.Callback callback) {
-        callback.invoke(origin, true, false);
-      }
-
-      protected void openFileChooser(ValueCallback<Uri> filePathCallback, String acceptType) {
-        getModule().startPhotoPickerIntent(filePathCallback, acceptType);
-      }
-      protected void openFileChooser(ValueCallback<Uri> filePathCallback) {
-        getModule().startPhotoPickerIntent(filePathCallback, "");
-      }
-      protected void openFileChooser(ValueCallback<Uri> filePathCallback, String acceptType, String capture) {
-        getModule().startPhotoPickerIntent(filePathCallback, acceptType);
-      }
-
-      @TargetApi(Build.VERSION_CODES.LOLLIPOP)
-      @Override
-      public boolean onShowFileChooser(WebView webView, ValueCallback<Uri[]> filePathCallback, FileChooserParams fileChooserParams) {
-        String[] acceptTypes = fileChooserParams.getAcceptTypes();
-        boolean allowMultiple = fileChooserParams.getMode() == WebChromeClient.FileChooserParams.MODE_OPEN_MULTIPLE;
-        Intent intent = fileChooserParams.createIntent();
-        return getModule().startPhotoPickerIntent(filePathCallback, intent, acceptTypes, allowMultiple);
-      }
-    });
+    webView.setWebChromeClient(new RNCWebChromeClient(reactContext, getModule(), webView));
```

Comment:
I'm definitely no Android expert but I was able to get this code into our project by taking this PR and the [Custom Android Guide](https://github.com/react-native-community/react-native-webview/blob/master/docs/Custom-Android.md) which worked really well!

The only thing that I had to change was change this `getModule` with `null` instead. I am really not sure why and how this works, but I figured to comment on it in case this might be relevant for others, or a potential bug.

The problem was basically that the `aPackage` was undefined in the `getModule` function which caused it to throw an error. As far as I can see that could have consequences for file upload.

Reply:
Hmm, I haven't had a problem with this in my example app. The file upload implementation is definitely different than what we were rolling with on our fork before we started to use `react-native-webview`. I would be glad to look into this more if I had a reproducible demo, though unless others have an issue with this too I don't believe it should be a blocker for merging this PR.

Reply:
This changed after some changes on master, and I've rebased and made this better. Now I no longer pass in the module here, and instead `getModule(reactContext)` from within the `RNCWebChromeClient` instance like is done elsewhere now. Shouldn't be a problem any longer.